lib/modulesync.rb

```diff
@@ -177,4 +178,29 @@ def self.update(cli_options)
     end
     exit 1 if errors && options[:fail_on_warnings]
   end
+
+  def self.execute(cli_options)
+    @options = config_defaults.merge(cli_options)
+
+    managed_modules.each do |puppet_module|
+      $stdout.puts "#{puppet_module.given_name}:"
+
+      puppet_module.repository.clone unless puppet_module.repository.cloned?
+      puppet_module.repository.switch branch: @options[:branch]
+
+      command_args = cli_options[:command_args]
+      local_script = File.expand_path command_args[0]
+      command_args[0] = local_script if File.exist?(local_script)
```

Comment:
I'm not a fan of this. I'd prefer to be more explicit in the behavior. If it's a path (on unix: it starts with `/`, `./` or `../`), only then would I treat it as a script. And `system()` can already do that. So personally I'd leave this out.

Reply:
Hum, as the _local script_ is relative to the top level (ie. `modulesync_config`) and no repository one, the `system()` command will fail. That's why I expanded the path before using it.

Reply:
Hmm, it does make sense but I can also see how it would be confusing. I'm not sure how to best deal with this. Perhaps document it somehow?

Reply:
Is my reply clear enough to resolve to conversation?

Reply:
I do wonder if the `--help` description for the command could explain this behavior but I don't consider it blocking.

Reply:
`msync help exec` now explain this behavior.